### ISSUE TITLE
`mamba create` with channel specified packages succeeds with missing packages

### ISSUE THREAD

**Issue by Anthchirp** [mamba-org/mamba#1054]
When creating an environment and specifying a non-existent package version with channel, then the creation will succeed without the package. An error message is shown, but the environment is still created with a `0` returncode. (I added `readline` so that there is something to install, but this is not required to demonstrate the issue. mamba will happily create an empty environment.)
```bash
 $ mamba create -p test-success --dry conda-forge::mamba=1001.1 conda-forge::readline

                  __    __    __    __
                 /  \  /  \  /  \  /  \
                /    \/    \/    \/    \
███████████████/  /██/  /██/  /██/  /████████████████████████
              /  / \   / \   / \   / \  \____
             /  /   \_/   \_/   \_/   \    o \__,
            / _/                       \_____/  `
            |/
        ███╗   ███╗ █████╗ ███╗   ███╗██████╗  █████╗
        ████╗ ████║██╔══██╗████╗ ████║██╔══██╗██╔══██╗
        ██╔████╔██║███████║██╔████╔██║██████╔╝███████║
        ██║╚██╔╝██║██╔══██║██║╚██╔╝██║██╔══██╗██╔══██║
        ██║ ╚═╝ ██║██║  ██║██║ ╚═╝ ██║██████╔╝██║  ██║
        ╚═╝     ╚═╝╚═╝  ╚═╝╚═╝     ╚═╝╚═════╝ ╚═╝  ╚═╝

        mamba (0.13.0) supported by @QuantStack

        GitHub:  https://github.com/mamba-org/mamba
        Twitter: https://twitter.com/QuantStack

█████████████████████████████████████████████████████████████


Looking for: ['conda-forge::mamba=1001.1', 'conda-forge::readline']

pkgs/main/linux-64       Using cache
pkgs/main/noarch         Using cache
pkgs/r/linux-64          Using cache
pkgs/r/noarch            Using cache
conda-forge/linux-64     Using cache
conda-forge/noarch       Using cache
ERROR   Selected channel specific (or force-reinstall) job, but package is not available from channel. Solve job will fail.
Transaction

  Prefix: /dls/tmp/wra62962/directories/SWlGMNS1/test-success

  Updating specs:

   - conda-forge::mamba==1001.1
   - conda-forge::readline


  Package        Version  Build        Channel                    Size
────────────────────────────────────────────────────────────────────────
  Install:
────────────────────────────────────────────────────────────────────────

  _libgcc_mutex      0.1  main         pkgs/main/linux-64         3 KB
  _openmp_mutex      4.5  1_gnu        pkgs/main/linux-64        22 KB
  libgcc-ng        9.3.0  h5101ec6_17  pkgs/main/linux-64         5 MB
  libgomp          9.3.0  h5101ec6_17  pkgs/main/linux-64       311 KB
  libstdcxx-ng     9.3.0  hd4cf53a_17  pkgs/main/linux-64         3 MB
  ncurses            6.2  he6710b0_1   pkgs/main/linux-64       817 KB
  readline           8.1  h46c0cb4_0   conda-forge/linux-64     Cached

  Summary:

  Install: 7 packages

  Total download: 9 MB

────────────────────────────────────────────────────────────────────────


DryRunExit: Dry run. Exiting.
```

This is unexpected, especially since
```bash
$ mamba create -p test-fail --dry -c conda-forge --override mamba=1001.1 readline

                  __    __    __    __
                 /  \  /  \  /  \  /  \
                /    \/    \/    \/    \
███████████████/  /██/  /██/  /██/  /████████████████████████
              /  / \   / \   / \   / \  \____
             /  /   \_/   \_/   \_/   \    o \__,
            / _/                       \_____/  `
            |/
        ███╗   ███╗ █████╗ ███╗   ███╗██████╗  █████╗
        ████╗ ████║██╔══██╗████╗ ████║██╔══██╗██╔══██╗
        ██╔████╔██║███████║██╔████╔██║██████╔╝███████║
        ██║╚██╔╝██║██╔══██║██║╚██╔╝██║██╔══██╗██╔══██║
        ██║ ╚═╝ ██║██║  ██║██║ ╚═╝ ██║██████╔╝██║  ██║
        ╚═╝     ╚═╝╚═╝  ╚═╝╚═╝     ╚═╝╚═════╝ ╚═╝  ╚═╝

        mamba (0.13.0) supported by @QuantStack

        GitHub:  https://github.com/mamba-org/mamba
        Twitter: https://twitter.com/QuantStack

█████████████████████████████████████████████████████████████


Looking for: ['mamba=1001.1', 'readline']

conda-forge/linux-64     Using cache
conda-forge/noarch       Using cache
Encountered problems while solving:
  - nothing provides requested mamba 1001.1**
```

Conda does fail as expected:
```
$ conda create -p test-fail-conda --dry conda-forge::mamba=1001.1 conda-forge::readline
Collecting package metadata (current_repodata.json): done
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed

PackagesNotFoundError: The following packages are not available from current channels:

  - conda-forge::mamba=1001.1

Current channels:

  - https://repo.anaconda.com/pkgs/main/linux-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/linux-64
  - https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```

**Comment by wolfv** [comment]
thanks for the bug report.

**Comment by adriendelsalle** [comment]
I just tried to reproduce and found another newly introduced bug, hopefully fixed by #1077.
With that fix I get the correct behavior.

Let us know @Anthchirp if it works for you when released:
